### ISSUE TITLE
Make the agent more resilient against DNS issues 

### ISSUE THREAD

**Issue by P6g9YHK6** [amidaware/tacticalrmm#1769]
**Is your feature request related to a problem? Please describe.**
When an agent loses DNS services it will not work anymore. (Mesh is not or less impacted)

**Describe the solution you'd like**
when a dns resolution fails it should fallback to the latest known working ip that was replied and probably changes the color of the "connectivity icon" to yellow or else.

**Additional context**
https://discord.com/channels/736478043522072608/1204337171633283092

**Comment by dinger1986** [comment]
could this not be simply accomplished by using a script which updates the hosts file?

**Comment by P6g9YHK6** [comment]
yes, but in this case i got the feeling that this is more a global "tactical" issue than a client sided one :)
for exemple the ESET management agent allow to set a list of whole lists of different addresses to access to the console. while my proposal does not go as far as this it is a common industry issue.

![image](https://github.com/amidaware/tacticalrmm/assets/17877371/1ec042ea-2e4d-4d12-9e3c-384c9bd06c2c)

**Comment by silversword411** [comment]
Why are you using port 4222?
Why aren't you using DNS?

**Comment by bbrendon** [comment]
LoL. I came here to say WTF. but I see the comment was deleted.
